pt_miniscreen/core/components/list.py

```diff
@@ -91,43 +110,40 @@ def _scroll_transition(self):
         self._rows_snapshot = None
         self.state.update({"active_transition": None, "transition_progress": 0})
 
-    def scroll_up(self):
+    def scroll_to(self, direction, distance=1):
```

Comment:
I'm not sure about this distance behaviour. It seems that if you scroll with a distance greater than 1 the order or self.rows no longer matches up with the Rows array and so it will behave pretty wierdly, particularly in the case where there are more than one visible row!

To scroll more the 1 row we should instantiate and add all the rows that need to be scrolled through and then remove the ones that are no longer visible when the transition ends.